Add WireframCuboid function to show oriented bounding boxes as computed ...

Jorge-C opened the pull request:
...from PCL.

davetcoleman commented:
You didn't add this function to the header file! Did you test it?

davetcoleman commented on src/rviz_visual_tools.cpp:
```diff
@@ -1077,6 +1077,58 @@ bool RvizVisualTools::publishPolygon(const geometry_msgs::Polygon &polygon, cons
+					       const Eigen::Matrix3d &rotation_matrix,
+					       const Eigen::Vector3d &min_point,
+					       const Eigen::Vector3d &max_point,
+					       const RvizVisualTools::colors &color=RvizVisualTools::BLUE)
```
these should be documented in the header file - why do i need both pose and position and rotation matrix?

Jorge-C replied:
I actually wanted to talk to you about this tomorrow, because I agree that pose is redundant for a general purpose cuboid drawing, but that's what we need to draw the pcl-computed bounding boxes at an arbitrary mesh pose. OTOH, maybe it's possible to take pcl's rotation matrix and position, put them together in an affine transform, and call `publishWireframeCuboid`with `pose * (rotation + position)`?

davetcoleman replied:
If this is custom for PCL, then I'd say keep a converter in your class, actually, and then make this function generic and more standard to how publishRectangle() works. but your custom function would call this one.

p.s. we should depreciate pubishRectangle and rename it Cuboid

davetcoleman commented on src/rviz_visual_tools.cpp:
```diff
@@ -1077,6 +1077,58 @@ bool RvizVisualTools::publishPolygon(const geometry_msgs::Polygon &polygon, cons
+					       const RvizVisualTools::colors &color=RvizVisualTools::BLUE)
+{
+  /* Publish cuboid at a certain pose. Useful to show an oriented
+     bounding as computed by PCL. */
```
no need to refer to PCL here

davetcoleman commented:
ping, i'd like to see this feature added!

Jorge-C commented:
@davetcoleman I've addressed your comments, how can I test this in an isolated way that doesn't interfere with anything else? I see a `.travis.yml` file in the repo root, does this have CI? Thanks!

davetcoleman commented:
Great!
No, Travis hasn't been upgraded to Ubuntu 14.04 so had to be turned off last Fall. 

I suggest you add a call to your function in the [publishTests()](https://github.com/davetcoleman/rviz_visual_tools/blob/development/include/rviz_visual_tools/rviz_visual_tools.h#L596) function and visualize it as [documented](https://github.com/davetcoleman/rviz_visual_tools/blob/development/README.md#quick-start)

davetcoleman commented on src/rviz_visual_tools.cpp:
```diff
@@ -1077,6 +1077,47 @@ bool RvizVisualTools::publishPolygon(const geometry_msgs::Polygon &polygon, cons
+                                               const Eigen::Matrix3d &rotation_matrix,
+                                               const Eigen::Vector3d &min_point,
+                                               const Eigen::Vector3d &max_point,
+                                               const RvizVisualTools::colors color=RvizVisualTools::BLUE)
```
Instead of specifying at position and rotation matrix, it should just pass in an ``Eigen::Affine3d`` that contains both, as is standard in all the other functions in this class.

Jorge-C replied:
Makes sense, do you know of a good eigen tutorial/how to combine rotation matrix & translation into `Affine3d`?

Jorge-C commented:
Ready for final review & merge!

davetcoleman commented:
Thanks!

Jorge-C commented:
Awesome, thanks for your help!
